### PR TITLE
Fix some outdated solver-interface and dimensions

### PR DESCRIPTION
**Pull request by MakisH**
I hope I have covered everything in the `pages/` directory in terms of:

- Removing `<solver-interface>`
- Moving the `dimensions` from the `solver-interface` to each `mesh`
- Checking that each `mesh` has a `dimensions`

### CODE REVIEW

File: pages/docs/couple-your-code/couple-your-code-gradient-data.md [hunk 1]
```diff
@@ -81,20 +81,17 @@ For the example, you can use the following `precice-config.xml` (note the versio
 ```xml
 <?xml version="1.0"?>
 
-<precice-configuration>
-
-  <solver-interface dimensions="3" experimental="on">
-
+<precice-configuration experimental="on">
```
Comment (IshaanDesai) [inline pages/docs/couple-your-code/couple-your-code-gradient-data.md]: Further below in the waveform section, it is `experimental="true"`. Lets be consistent with the value of the tag. Also, is coupling gradient data still experimental?

File: pages/docs/configuration/configuration-basics.md [hunk 2]
```diff
@@ -56,12 +50,14 @@ int temperatureID = precice.getDataID("Temperature", meshID);
...
 
 ```xml
-<mesh name="MyMesh1"> 
+<mesh name="MyMesh1" dimensions="3"> 
   <use-data name="Temperature"/> 
   <use-data name="Forces"/> 
 </mesh> 
 ```
 
+The value `dimensions` needs to match the physical dimension of your simulation, i.e. the number of coordinates a vertex has in `setMeshVertex`, etc. Some solvers only support 3D simulation, such as OpenFOAM or CalculiX. In this case the adapter maps from 3D to 2D if the preCICE dimension is 2D. This, of course, only works if you simulate a quasi-2D scenario with one layer of cells in z direction.
```
Comment (IshaanDesai) [inline pages/docs/configuration/configuration-basics.md:59]: Is the Z direction hard-coded as the dead direction?

**Review by IshaanDesai — APPROVED** [review]
Looks good, just two minor comments.

File: pages/docs/configuration/configuration-basics.md (hunk 2 above)
Comment (MakisH) [reply pages/docs/configuration/configuration-basics.md:59]: This is an example for the OpenFOAM adapter, in which yes, the Z (third index) is hard-coded to be the dead direction.

File: pages/docs/couple-your-code/couple-your-code-gradient-data.md (hunk 1 above)
Comment (MakisH) [reply pages/docs/couple-your-code/couple-your-code-gradient-data.md]: This PR is only moving things around, but good catch anyway! :D Fixed.

**Pull request merged**